data/frame_test.go

```diff
@@ -48,6 +48,235 @@ func ExampleNewFrame() {
 	// +-------------------------------+-----------------------+-----------------------+
 }
 
+type mockPoint struct {
+	Time  time.Time
+	Value float64
+}
+
+type mockSeries struct {
+	Name   string
+	Labels map[string]string
+	Points []mockPoint
+}
+
+type mockResponse struct {
+	Series []mockSeries
+}
+
+func ExampleFrame_tSDBTimeSeriesDifferentTimeIndices() {
+	// A common tsdb response pattern is to return a collection
+	// of time series where each time series is uniquely identified
+	// by a Name and a set of key value pairs (Labels (a.k.a Tags)).
+
+	// In the case where the responses does not share identical time values and length (a single time index),
+	// then the proper representation is a []*Frame. Where each Frame has a Time type field and one or more
+	// Number fields.
+
+	// Each Frame should have its value sorted by time in ascending order.
+
+	res := mockResponse{
+		[]mockSeries{
+			mockSeries{
+				Name:   "cpu",
+				Labels: map[string]string{"host": "a"},
+				Points: []mockPoint{
+					{
+						time.Date(2020, 1, 2, 3, 4, 0, 0, time.UTC), 3,
+					},
+					{
+						time.Date(2020, 1, 2, 3, 5, 0, 0, time.UTC), 6,
+					},
+				},
+			},
+			mockSeries{
+				Name:   "cpu",
+				Labels: map[string]string{"host": "b"},
+				Points: []mockPoint{
+					{
+						time.Date(2020, 1, 2, 3, 4, 1, 0, time.UTC), 4,
+					},
+					{
+						time.Date(2020, 1, 2, 3, 5, 1, 0, time.UTC), 7,
+					},
+				},
+			},
+		},
+	}
+
+	frames := make([]*data.Frame, len(res.Series))
```

Comment:
Good write up! I think we might look into helper functions for this. But it's not critical

Reply:
Good point, also true of `map[string]*data.DataResponse` (or something like that) in `backend`, will do those together.

Reply:
Great :+1: